crates/uv-distribution/src/source/mod.rs

```diff
@@ -966,16 +966,35 @@ impl<'a, T: BuildContext> SourceDistCachedBuilder<'a, T> {
     async fn build_source_dist_metadata(
         &self,
         dist: &SourceDist,
-        source_dist: &Path,
+        source_tree: &Path,
         subdirectory: Option<&Path>,
     ) -> Result<Option<Metadata21>, Error> {
         debug!("Preparing metadata for: {dist}");
 
+        // Attempt to read static metadata from the source distribution.
+        match read_pkg_info(source_tree).await {
+            Ok(metadata) => {
+                debug!("Found static metadata for: {dist}");
+
+                // Validate the metadata.
+                if &metadata.name != dist.name() {
+                    return Err(Error::NameMismatch {
+                        metadata: metadata.name,
+                        given: dist.name().clone(),
+                    });
+                }
+
+                return Ok(Some(metadata));
+            }
+            Err(Error::MissingPkgInfo | Error::DynamicPkgInfo(_)) => {}
```

Comment:
Could you also add a debug log here, so users running with `-v` can figure out why this source dist takes the slow path